HOTT-846: Add Rules of Origin API client

jebw opened the pull request:
### Jira link

[HOTT-846](https://transformuk.atlassian.net/browse/HOTT-846)

### What?

I have added/removed/altered:

- [x] Added an API client for the `/rules_of_origin_schemes` endpoint
- [x] Disabled feature flag in `Rails.env.development?` - the logic being other backend branches may not yet have latest api changes - this avoids `master` frontend having a hard dependency on tip of `master` backend. Developers can enable rules of origin by setting feature flag locally.

### Why?

I am doing this because:

- We should be querying the backend rather then showing mocked data from the frontend

### Notes

This supercedes #195

jebw committed:
HOTT-846: Added Rules of Origin api client

jebw committed:
HOTT-846: Use new Schemes based API endpoint

willfish commented on app/models/goods_nomenclature.rb:
```diff
@@ -51,4 +51,14 @@ def id
   end
+
+  def rules_of_origin(*args, **kwargs)
+    return nil unless try(:declarable?)
```
Don't we always have this answer?

jebw replied:
Not on `Chapter` which also inherits from `GoodsNomenclature`

willfish commented on spec/models/goods_nomenclature_spec.rb:
```diff
@@ -92,4 +92,72 @@
+
+  describe '#rules_of_origin' do
+    before do
+      # Needed because goods nomenclature isn't declarable but #rules_of_origin
```
I think we can reflect that section, chapters, etc, all have declarable and just treat them as ducks/trust the data

jebw replied:
Ok, will change `Chapter` to always return false for `#declarable?`

jebw replied:
This override will still be required though because we want to bypass the short-circuit

willfish reviewed (approved):
Love it

jebw committed:
Set declarable to be false for GoodsNomenclature

Inheriting classes can override if appropriate - Heading and Commodity already do

jebw committed:
HOTT-846: Dropped rules API call support

Since there's not a direct rules api endpoint anymore
